apps/react-editor/index.d.ts

```diff
@@ -2,7 +2,13 @@ import { Component } from 'react';
 import ToastuiEditor, { EditorOptions } from '@toast-ui/editor';
 import ToastuiEditorViewer, { ViewerOptions } from '@toast-ui/editor/dist/toastui-editor-viewer';
 
-type EditorProps = Omit<EditorOptions, 'el'>;
+type EditorProps = Omit<EditorOptions& {
+  onLoad?: (param: Editor) => void;
+  onChange?: (param: { source: SourceType | 'viewer'; data: MouseEvent }) => void;
+  onStateChange?: (param: MarkdownToolbarState | WysiwygToolbarState) => void;
+  onFocus?: (param: { source: SourceType }) => void;
+  onBlur?: (param: { source: SourceType }) => void;
```

Comment:
`SourceType`, `MarkdownToolbarState`, `WysiwygToolbarState` Where did you get these types from?

Reply:
> `SourceType`, `MarkdownToolbarState`, `WysiwygToolbarState` Where did you get these types from?

It is from apps/editor/index.d.ts 191
```
  interface EventMap {
    load?: (param: Editor) => void;
    change?: (param: { source: SourceType | 'viewer'; data: MouseEvent }) => void;
    stateChange?: (param: MarkdownToolbarState | WysiwygToolbarState) => void;
    focus?: (param: { source: SourceType }) => void;
    blur?: (param: { source: SourceType }) => void;
  }
```

The doc said that in React component, the user could just add *on* before the event name to pass callback funtion. And I 
 speculate that the types is similiar to the types in the main editor.

Reply:
I wish there were some more elegant ways to do it, but it seems I have no choice but to do so.

Reply:
@yar2001 Sorry for the late answer. You can `import` the editor type, but when I check the [Editor's type declaration file](https://github.com/nhn/tui.editor/blob/master/apps/editor/index.d.ts),  it is not exported. If possible, how about modifying the editor's declaration file as well?

For example, you can modify the `SourceType` type as follows. You can process other types like this.

> e.g. https://github.com/nhn/tui.editor/blob/301f5e6ac36b75dfc6bb40eb35fcb3b474fb8142/apps/editor/index.d.ts#L189

```js
- type SourceType = 'wysiwyg' | 'markdown';
+ export type SourceType = 'wysiwyg' | 'markdown';
```